internal/gatewayapi/tls.go

```diff
@@ -54,8 +67,95 @@ func validateTLSSecretData(secret *corev1.Secret) error {
 			parseErr = fmt.Errorf("unable to parse EC formatted private key in %s", corev1.TLSPrivateKeyKey)
 		}
 	default:
-		return fmt.Errorf("%s key format found in %s, supported formats are PKCS1, PKCS8 or EC", keyBlock.Type, corev1.TLSPrivateKeyKey)
+		return publicKeyAlgorithm, certFQDN, fmt.Errorf("%s key format found in %s, supported formats are PKCS1, PKCS8 or EC", keyBlock.Type, corev1.TLSPrivateKeyKey)
+	}
+
+	return publicKeyAlgorithm, certFQDN, parseErr
+}
+
+func getFQDN(cert *x509.Certificate) string {
+	fqdn := ""
+	if len(cert.DNSNames) > 0 {
+		for _, name := range cert.DNSNames {
+			fqdn += name + ","
+		}
+	} else {
+		fqdn = cert.Subject.CommonName
+	}
+	return fqdn
+}
+
+// https://golang.google.cn/doc/go1.15#commonname
+func verifyHostname(cert *x509.Certificate, host string) error {
+	lowered := toLowerCaseASCII(host)
+	if len(cert.DNSNames) > 0 {
+		for _, match := range cert.DNSNames {
+			if matchHostnames(toLowerCaseASCII(match), lowered) {
```

Comment:
I don't think so, as this is checking for a hostname in the x509.Certificate, first checking if SAN extension is present, then checking  Subject CommonName field.

I think computeHosts would be valuable if we wanted to get hosts from both listener and routes instead of this: 
https://github.com/envoyproxy/gateway/blob/1ff177fd3bde7ccb31cb559c0350d7cdc84de3da/internal/gatewayapi/validate.go#L340

Reply:
I meant as a replacement for `matchHostnames` , they both seem to doing the same thing, ensuring that the list of strings in ARG1, must be a subset of ARG2

Reply:
I will check it then

Reply:
Thanks for pointing me to this function :)